src/scripts/client/modules/hw.js

```diff
@@ -0,0 +1,200 @@
+( function( tiy, moment ) {
+  'use strict';
+
+  const HW_NO_CLIENT_TEMPLATE = 'templates/hw.html';
+
+  let $ui = null;
+  let pageData = {};
+
+  tiy.loadModule( {
+    name: 'hw',
+    navIcon: 'fa-download',
+    render: main
+  } );
+
+  function main( data, elem ) {
+    $ui = $( elem );
+    pageData = data;
+
+    // On relevant pages add links
+    if ( pageData.assignment || pageData.assignment_submission ) {
+      addCloneLinksToUi();
+    }
+
+    $( $ui ).on( 'showing', function() {
+      $ui.append( `<i class='newline_hw_working fa fa-spin fa-spinner'></i>` );
+
+      detectNativeClient( function handleNativeClient( resp ) {
+        addMainContent( $ui, resp );
+      } );
+    } );
+  }
+
+
+  function addMainContent( $ui, resp ) {
+    $ui
+      .html( "" )
+      .append( `<h5>Newline HW ${resp.status === "ok" ? 'detected' : 'not found'}</h5>` );
+
+    if ( resp.status === "ok" ) {
+      $ui.append( `Last Heartbeat at ${moment(resp.message_at).fromNow()}` );
+      $ui.append( `<dl>` );
+
+      for ( let i in resp.data ) {
```

Comment:
I might rewrite this as:

```
Object.keys(resp.data).forEach(function showData(key) {
  $ui.append( `<dt>${key}</dt><dd>${resp.data[key]}</dd>` );
});
```

(`Object.keys()` won't pull in prototype properties.)